util.go

```diff
@@ -2,24 +2,46 @@ package cuckoo
 
 import (
 	"encoding/binary"
-	"math/rand"
+	"unsafe"
 
-	metro "github.com/dgryski/go-metro"
+	"github.com/zeebo/wyhash"
+	"github.com/zeebo/xxh3"
+)
+
+var (
+	// plus 1 to prevent out of range.
+	altHash = [maxFingerprint + 1]uint{}
```

Comment:
This comes with a memory penalty of 2^16 /1024 = 64kb.

Please do a separate PR/evaluation to see if it's worth.

Reply:
Ok, I agree with your argument that in most cases, the filter itself will already be quite big, hence the additional 64 kb don't add much. Please send as separate pull request.